src/lib/core/components/filter/HistogramFilter.tsx

```diff
@@ -80,28 +79,32 @@ export function HistogramFilter(props: Props) {
 
     if (variable.type === 'number')
       return {
-        binWidth: variable.binWidthOverride ?? variable.binWidth,
+        binWidth: variable.binWidthOverride ?? variable.binWidth ?? 0.1,
         binWidthTimeUnit: undefined,
         independentAxisRange:
           variable.displayRangeMin != null && variable.displayRangeMax != null
             ? { min: variable.displayRangeMin, max: variable.displayRangeMax }
-            : undefined,
+            : { min: Math.min(0, variable.rangeMin!), max: variable.rangeMax! },
         ...otherDefaults,
       };
 
     // else date variable
-    const binWidthString = variable.binWidthOverride ?? variable.binWidth;
-    const binWidth = binWidthString
-      ? parseTimeDelta(binWidthString)
-      : undefined;
+    const binWidth = variable.binWidthOverride ?? variable.binWidth;
+    const binUnits = variable.binUnits;
 
     return {
-      binWidth: binWidth?.value,
-      binWidthTimeUnit: binWidth?.unit as TimeUnit, // bit nasty!
+      binWidth: binWidth ?? 1,
+      binWidthTimeUnit: binUnits ?? variable.binUnits!, // bit nasty!
       independentAxisRange:
         variable.displayRangeMin != null && variable.displayRangeMax != null
-          ? { min: variable.displayRangeMin, max: variable.displayRangeMax }
-          : undefined,
+          ? {
+              min: variable.displayRangeMin + 'T00:00:00',
+              max: variable.displayRangeMax + 'T00:00:00',
+            }
+          : {
+              min: variable.rangeMin! + 'T00:00:00',
```

Comment:
If `variable.rangeMin` is missing, `'T00:00:00'` is likely to be an invalid date for the back end.  

See previous comment above.